clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -238,6 +246,14 @@ private void completeFutureAndFireCallbacks(long baseOffset, long logAppendTime,
         produceFuture.done();
     }
 
+    public List<ProducerBatch> getChildrenProducerBatch() {
+        return childrenProducerBatch;
+    }
+
+    public void clearChildrenProducerBatch() {
+        childrenProducerBatch.clear();
```

Comment:
I'm wondering if we should take care of concurrent modification for this list.

Reply:
Good point,  knowing the ArrayList is not thread-safe, not sure if going to throw an Exception, I will sync the access to that method, thanks for the comment.

Reply:
Done